dbt/models/marts/students/dim_student_surveys.sql

```diff
@@ -0,0 +1,104 @@
+-- model: dim_student_surveys 
+
+with 
+contained_levels as (
+    select * 
+    from {{ ref('stg_dashboard__contained_levels') }}
+),
+
+contained_levels_answers as (
+    select distinct 
+        level_id, answer_number, answer_text
+    from {{ ref('stg_dashboard__contained_level_answers') }}
+),
+
+levels as (
+    select 
+        *,   
+        case 
+            when level_type = 'multi' then json_array_length(json_extract_path_text(properties, 'answers'))
+            when level_type = 'freeresponse' 	then 1 
+            when level_type = 'external'		then 0 
+        end 	as num_response_options 
+
+    from {{ ref('stg_dashboard__levels') }}
+),
+
+script_levels as (
+    select *
+    from {{ ref('stg_dashboard__script_levels') }}
+),
+
+levels_script_levels as (
+    select * 
+    from {{ ref('stg_dashboard__levels_script_levels') }}
+),
+
+scripts as (
+    select * 
+    from {{ ref('stg_dashboard__scripts') }}   
+),
+
+course_structure as (
+    select * 
+    from {{ ref('dim_course_structure') }}
+),
+
+combined as (
+    select 
+        gl.level_id                     as group_level_id,
+        lower(gl.level_name)            as survey_name,
+        
+        case 
+            when survey_name like '%pre%' 
+             and survey_name not like '%preview%'  then 'pre'
+            when survey_name like '%post%'         then 'post'
+            when survey_name like '%pulse%'        then 'pulse'
+            when survey_name like '%end of unit%'  then 'end of unit'
+            else null end              as survey_type,
+        
+        sc.script_name,
+        cl.level_id                     as contained_level_id,
+        cl.level_name                   as question_name,
+        cl.level_type                   as question_type,
+        col.contained_level_position    as question_position,
+        col.contained_level_text        as question_text,
+        
+        cl.num_response_options,
+        cola.answer_text        as response_option,
+        cola.answer_number      as response_position,
```

Comment:
```suggestion
        cola.answer_number,    
```
The answer_number is not the response position, it is the numerical value of the answer. For example, in a 7-point scale of Agree/Disagree 7 is the highest level of agreement, and 1 the lowest. This is used to calculate averages, so labeling it as the position would confuse users.